src/raft.cpp

```diff
@@ -22,30 +22,53 @@ void Raft::generate(SliceDataStorage& storage)
     const coord_t distance = settings.get<coord_t>("raft_margin");
     constexpr bool include_support = true;
     constexpr bool include_prime_tower = true;
-    storage.raftOutline = storage.getLayerOutlines(0, include_support, include_prime_tower).offset(distance, ClipperLib::jtRound);
+    Polygons global_raft_outlines{ storage.getLayerOutlines(0, include_support, include_prime_tower).offset(distance, ClipperLib::jtRound) };
     const coord_t shield_line_width_layer0 = settings.get<coord_t>("skirt_brim_line_width");
     if (storage.draft_protection_shield.size() > 0)
     {
         Polygons draft_shield_raft = storage.draft_protection_shield.offset(shield_line_width_layer0) // start half a line width outside shield
                                         .difference(storage.draft_protection_shield.offset(-distance - shield_line_width_layer0 / 2, ClipperLib::jtRound)); // end distance inside shield
-        storage.raftOutline = storage.raftOutline.unionPolygons(draft_shield_raft);
+        global_raft_outlines = global_raft_outlines.unionPolygons(draft_shield_raft);
     }
     if (storage.oozeShield.size() > 0 && storage.oozeShield[0].size() > 0)
     {
         const Polygons& ooze_shield = storage.oozeShield[0];
         Polygons ooze_shield_raft = ooze_shield.offset(shield_line_width_layer0) // start half a line width outside shield
                                         .difference(ooze_shield.offset(-distance - shield_line_width_layer0 / 2, ClipperLib::jtRound)); // end distance inside shield
-        storage.raftOutline = storage.raftOutline.unionPolygons(ooze_shield_raft);
+        global_raft_outlines = global_raft_outlines.unionPolygons(ooze_shield_raft);
     }
 
     if(settings.get<bool>("raft_remove_inside_corners"))
     {
-        storage.raftOutline.makeConvex();
+        global_raft_outlines.makeConvex();
     }
     else
     {
         const coord_t smoothing = settings.get<coord_t>("raft_smoothing");
-        storage.raftOutline = storage.raftOutline.offset(smoothing, ClipperLib::jtRound).offset(-smoothing, ClipperLib::jtRound); // remove small holes and smooth inward corners
+        global_raft_outlines = global_raft_outlines.offset(smoothing, ClipperLib::jtRound).offset(-smoothing, ClipperLib::jtRound); // remove small holes and smooth inward corners
+    }
+
+    constexpr bool dont_include_prime_tower = false;
+    Polygons raw_raft_without_prime{ storage.getLayerOutlines(0, include_support, dont_include_prime_tower).offset(distance, ClipperLib::jtRound) };
```

Comment:
Could we maybe use this one to start with, and only combine in `global_raft_outlines` with `storage.primeTower.outer_poly_first_layer`? Would prevent calling `getLayerOutlines` multiple times and doing that offset multiple times.